ymtdzzz/otel-tui
# feat: zipkin support and option to enable it

related: #125

Adding zipkin receiver to be able to receive zipkin traces.

![image](https://github.com/user-attachments/assets/4ead77cb-027a-47eb-b39d-13e49093aaae)

# How to test
Build otel-tui as a docker image

```
$ docker build . -t otel-tui:test
```

Check out the example repository: https://github.com/openzipkin/brave-example

Modify `docker-compose.yml` in brave-example directory to use the `otel-tui:test` container instead of `zipkin`

```yaml
# permit depends_on/condition: service_healthy
version: "2.4"

# BRAVE_EXAMPLE choices are listed here https://github.com/openzipkin/brave-example#example-projects

services:
  # Generate traffic by hitting http://localhost:8081
  frontend:
    container_name: frontend
    image: ghcr.io/openzipkin/brave-example:${BRAVE_EXAMPLE:-armeria}
    entrypoint: start-frontend
    ports:
      - 8081:8081
    depends_on:
      backend:
        condition: service_healthy
    #  zipkin:
    #    condition: service_healthy
  # Serves the /api endpoint the frontend uses
  backend:
    container_name: backend
    image: ghcr.io/openzipkin/brave-example:${BRAVE_EXAMPLE:-armeria}
    entrypoint: start-backend
    # depends_on:
    #   zipkin:
    #     condition: service_healthy
  # View traces at http://127.0.0.1:9411/zipkin
  zipkin:
    #image: ghcr.io/openzipkin/zipkin-slim
    image: otel-tui:test
    container_name: zipkin
    ports:
      - 9411:9411
    stdin_open: true
    tty: true
    entrypoint: ["/otel-tui", "--enable-zipkin"]
```

Run this example

```sh
docker compose up -d
```

In another terminal session, attach to otel-tui container (note that the container name is still `zipkin`)

```
$ docker compose attach zipkin
```

Access to the example endpoint: http://localhost:8081/

Check that the traces send to otel-tui and processed correctly!
